website/docs/Queries.md

```diff
@@ -391,6 +392,23 @@ const element = screen.getByA11yValue({ now: 25 });
 const element2 = screen.getByA11yValue({ text: /25/ });
 ```
 
+### `ByPredicate`
+
+> getByPredicate, getAllByPredicate, queryByPredicate, queryAllByPredicate, findByPredicate, findAllByPredicate
+
+```ts
+getByPredicate(
+  predicate: (element: ReactTestInstance) => boolean,
+  options?: {
+    includeHiddenElements?: boolean;
+  }
+): ReactTestInstance;
+```
+
+Returns a host element matching a custom `predicate` function.
+
+This query type is an escape hatch and should be used with care. In most cases you using the standard queries like `getByRole` or `getByText` will lead to test more-resembling user perspective. Use this query with care in rare cases where more flexibility is needed.
```

Comment:
could we provide an example of a real scenario where we might need it and how to use it in that case? (like it's done for ByAccessibilityValue above)

Reply:
i see what you mean, but then RNTL only provides queries for common case scenarios. We could find a legit scenario  that's very uncommon maybe? Like find all underline text? (so hard to find a rare but legit scenario haha ^^')

Reply:
Yeah, we can build examples to showcase how you can recreate some existing queries like *ByTestId.

I went back to the source, [DTL docs](https://testing-library.com/docs/dom-testing-library/api-custom-queries), to check what examples they give, but they are basically implementing `*ByTestId` queries using some `queryByAttribute` function. All that seems like legacy stuff that has no good examples.